### ISSUE TITLE
Support "tvg-url"

### ISSUE THREAD

**Issue by xmha97** [kodi-pvr/pvr.iptvsimple#308]
Hello, 

Every channel has a separate EPG
for example:
http://epg.me/tv1.xml
http://epg.me/tv2.xml
http://epg.me/tv3.xml

Please support "tvg-url"

```
#EXTINF:-1 tvg-logo="https://logo.com/TV1.png" tvg-url="http://epg.me/tv1.xml" tvg-id="TV1" tvg-name="TV1",TV1
https://tv1.com/steam.m3u8
#EXTINF:-1 tvg-logo="https://logo.com/TV2.png" tvg-url="http://epg.me/tv2.xml" tvg-id="TV2" tvg-name="TV2",TV2
https://tv2.com/steam.m3u8
#EXTINF:-1 tvg-logo="https://logo.com/TV3.png" tvg-url="http://epg.me/tv3.xml" tvg-id="TV3" tvg-name="TV3",TV3
https://tv3.com/steam.m3u8
```

**Comment by phunkyfish** [comment]
This is not possible. The addon currently only supports a single XMLTV file.

**Comment by phunkyfish** [comment]
I’ve been thinking of this a bit more. How would you see it working?

Let’s say a user has specified an XMLTV file in the addon settings. Would you say only use the per channel files if the user has not specified one?

**Comment by phunkyfish** [comment]
What’s the content of a channel specific epg URL? 

Do you have a sample file you can attach?

**Comment by daker** [comment]
@phunkyfish 

We can have a case where the egp comes from multiple source, and since it was not supported i had to make a middleware to combine those into one xml.

An other issue is when the xml is big, the parsing can take time in kodi, for ex the xml tv for 300 french channels is 80M https://www.xmltv.fr/guide/tvguide.xml , so i guess doing things in small chunks would help.

**Comment by phunkyfish** [comment]
80MB for the epg for one channel? Sorry for 300 channels. So how can each channel have a tvg-url? It doesn’t make sense.

For the XMLTV file to be valid it must contain the channel entry and the epg entries only for that one channel.

Smaller chunks won’t help, it will take the same amount of time to load.

**Comment by Leatherface75** [comment]
Here is an example of xmltv service with xmltv files for each channel.

http://xmltv.xmltv.se/

**Comment by phunkyfish** [comment]
Yes, and you would tend to aggregate these files using a tool like webgrab into valid XMLTV files. These files on there own are not valid and there is no standard for referencing them (de facto or otherwise) from an M3U file.

**Comment by Leatherface75** [comment]
Yeah that's what i am doing with TVheadend and xmltv-grabbers package.
I just mention it in case someone want it the way asked here.
